docs/sources/flow/reference/components/pyroscope.java.md

```diff
@@ -0,0 +1,177 @@
+---
+aliases:
+  - /docs/grafana-cloud/agent/flow/reference/components/pyroscope.java/
+  - /docs/grafana-cloud/monitor-infrastructure/agent/flow/reference/components/pyroscope.java/
+  - /docs/grafana-cloud/monitor-infrastructure/integrations/agent/flow/reference/components/pyroscope.java/
+  - /docs/grafana-cloud/send-data/agent/flow/reference/components/pyroscope.java/
+canonical: https://grafana.com/docs/agent/latest/flow/reference/components/pyroscope.java/
+description: Learn about pyroscope.java
+title: pyroscope.java
+---
+
+# pyroscope.java
+
+{{< docs/shared lookup="flow/stability/beta.md" source="agent" version="<AGENT_VERSION>" >}}
+
+`pyroscope.java` continuously profiles Java processes running on the local Linux OS
+using [async-profiler](https://github.com/async-profiler/async-profiler).
+
+{{% admonition type="note" %}}
+To use the  `pyroscope.java` component you must run {{< param "PRODUCT_NAME" >}} as root and inside host pid namespace.
+{{% /admonition %}}
+
+## Usage
+
+```river
+pyroscope.java "LABEL" {
+  targets    = TARGET_LIST
+  forward_to = RECEIVER_LIST
+}
+```
+
+## Arguments
+
+The following arguments are supported:
+
+| Name         | Type                     | Description                                      | Default | Required |
+|--------------|--------------------------|--------------------------------------------------|---------|----------|
+| `targets`    | `list(map(string))`      | List of java process targets to profile.         |         | yes      |
+| `forward_to` | `list(ProfilesReceiver)` | List of receivers to send collected profiles to. |         | yes      |
+| `tmp_dir`    | `string`                 | Temporary directory to store async-profiler      | `/tmp`  | no       |
+
+## Profiling behavior
+
+The special label `__process_pid__` _must always_ be present in each target of `targets` and corresponds to the `PID` of
+the process to profile.
+
+Upon component startup, `pyroscope.java` will create a temporary directory under `tmp_dir` and extract the
+async-profiler binaries for both glibc and musl into it with the following layout.
+
+```
+/tmp/grafana-agent-asprof-glibc-{SHA1}/bin/asprof
+/tmp/grafana-agent-asprof-glibc-{SHA1}/lib/libasyncProfiler.so
+/tmp/grafana-agent-asprof-musl-{SHA1}/bin/asprof
+/tmp/grafana-agent-asprof-musl-{SHA1}/lib/libasyncProfiler.so
+```
+
+Upon process profiling startup, the component detects libc type and copies according `libAsyncProfiler.so` into the
+target process file system at the exact same path.
+
+{{% admonition type="note" %}}
+The `asprof` binary is executed as privileged user.
```

Comment:
```suggestion
The `asprof` binary must be executed by a privileged user.
```

Reply:
just to clarify: it is the  `pyroscope.java` compoenents which executes `asprof` binary.

`pyroscope.java` should be run as privileged. Users do not execute `asprof` manually.

Is suggestion aligned with it?

Reply:
cc @clayton-cornell

Reply:
@korniltsev Ah that wasn't clear to me from the context. 

What about something like this:

```suggestion
The `asprof` binary runs with elevated permissions.
```